iGlance/iGlance/iGlance/MenuBarItems/DiskUsageMenuBarItem.swift

```diff
@@ -0,0 +1,155 @@
+//  Copyright (C) 2020  D0miH <https://github.com/D0miH> & Contributors <https://github.com/iglance/iGlance/graphs/contributors>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import Foundation
+import CocoaLumberjack
+
+class DiskUsageMenuBarItem: MenuBarItem {
+    // MARK: -
+
+    func update() {
+        updateMenuBarIcon()
+    }
+
+    /**
+     * Updates the icon of the menu bar item. This function is called during every update interval.
+     */
+    func updateMenuBarIcon() {
+        // get the button of the menu bar item
+        guard let button = self.statusItem.button else {
+            DDLogError("Could not retrieve the button of the 'NetworkMenuBarItem'")
+            return
+        }
+
+        // get the sizes
+        let (used, free) = DiskInfo.getFreeDiskSize()
+        let usedSpace = convertToCorrectUnit(bytes: used)
+        let freeSpace = convertToCorrectUnit(bytes: free)
+
+        let menuBarImage = createMenuBarImage(up: usedSpace, down: freeSpace)
+
+        // set the menu bar item image
+        button.image = menuBarImage
+    }
+
+    // MARK: -
+    // MARK: Private Functions
+
+    /**
+    *  Takes the amount of space in bytes and returns the correct value according to the unit as a string and the correct unit (KB, MB, GB, TB) as a string.
+    *  If the given value of bytes is smaller than 1000 the function will return a value as is and unit of "B".
+    *
+    *  - Parameter bytes: The given number of bytes
+    *
+    *      Examples:
+    *          512 Bytes -> (value: "512", unit: "B")
+    *          5_000 Bytes -> (value: "5", unit: "KB")
+    *          5_000_000 Bytes -> (value: "5", unit: "MB")
+    *          5_000_000_000 Bytes -> (value: "5", unit: "GB")
+    *
+    */
+    func convertToCorrectUnit(bytes: Int) -> (value: String, unit: String) {
+        if bytes < 1000 {
+            return (value: "\(bytes)", unit: "B")
+        }
+        let exp = Int(log2(Double(bytes)) / log2(1000.0))
```

Comment:
Wow, very elegant!

Reply:
Oh I’ve found this elegant solution on stackoverflow :)